src/services/getEditsForFileRename.ts

```diff
@@ -1,67 +1,159 @@
+            if (!pathIsRelative(importText) ||
+                importIsInternalToDirectory(oldFileOrDirPath, sourceFile.fileName, importText) ||
+                importIsInternalToDirectory(newFileOrDirPath, sourceFile.fileName, importText)) {
+                return undefined;
             }
-        }
-        return result;
+            // If we're renaming "./a" to "./foo/a", an import from "./foo/z" can just import "./a".
+            const short = tryRemovePrefix(importText, relativeOldDirToNewDir + "/");
+            if (short !== undefined) return ensurePathIsNonModuleName(short);
+            return combineNormal(relativeNewDirToOldDir, importText);
+        };
     }
 
-    function getPathUpdater(oldFilePath: string, newFilePath: string, host: LanguageServiceHost): (oldPath: string) => string | undefined {
+    function getRelative(from: string, to: string): string {
+        return ensurePathIsNonModuleName(getRelativePathFromDirectory(toDirectory(from), toDirectory(to), /*ignoreCase*/ false));
+    }
+
+    function toDirectory(path: string): string {
+        return isAnySupportedFileExtension(path) ? getDirectoryPath(path) : path;
+    }
+
+    function importIsInternalToDirectory(oldDir: string, sourceFilePath: string, importText: string): boolean {
+        return !!removeDirectoryPrefixFromPath(oldDir, combineNormal(getDirectoryPath(sourceFilePath), importText));
+    }
+
+    /**
+     * Path updater for imports external to the file(s) being moved -- we will update these if they imported from the moved file.
+     * E.g., an import from "./old" will need to import from "./new".
+     *
+     * @param oldFullPath Absolute path to a failed lookup location
+     * @param oldRelPath Actual import text
+     */
+    type ExternalPathUpdater = (oldFullPath: string, oldRelPath: string, isImport: boolean) => string | undefined;
+    function getExternalPathUpdater(oldFileOrDirPath: string, newFileOrDirPath: string, host: LanguageServiceHost): ExternalPathUpdater {
         // Get the relative path from old to new location, and append it on to the end of imports and normalize.
-        const rel = getRelativePathFromFile(oldFilePath, newFilePath, createGetCanonicalFileName(hostUsesCaseSensitiveFileNames(host)));
-        return oldPath => {
-            if (!pathIsRelative(oldPath)) return;
-            return ensurePathIsNonModuleName(normalizePath(combinePaths(getDirectoryPath(oldPath), rel)));
+        const rel = getRelativePathFromFile(oldFileOrDirPath, newFileOrDirPath, createGetCanonicalFileName(hostUsesCaseSensitiveFileNames(host)));
+        return (fullPath, importText, isImport) => {
+            if (fullPath === oldFileOrDirPath) {
+                const newImportText = pathIsRelative(importText)
+                    ? combinePathsSafe(tryRemoveIndexOrPackage(fullPath) && !endsWithSomeIndex(importText) ? importText : getDirectoryPath(importText), rel)
+                    : newFileOrDirPath;

```

Comment:
With amd resolution, renaming /a/folder/oldModule.ts to /a/folder/newModule.ts and if I had file as
```typescript
// /a/folder/someApp.ts 
import { a } from "oldModule"
``` 
to change to 
```typescript
// /a/folder/someApp.ts 
import { a } from "/a/folder/newModule"
```